src/Components/Tooltip.tsx

```diff
@@ -0,0 +1,83 @@
+import React from "react"
+
+import styled from "styled-components"
+import colors from "../Assets/Colors"
+
+interface Props extends React.HTMLProps<HTMLSpanElement> {
+  message: string
+}
+
+const TooltipContainer = styled.span`
+  display: inline-block;
+  position: relative;
+  cursor: help;
+  margin: 0 0.5em;
+  width: 14px;
+  height: 14px;
+  margin-bottom: -2px;
+
+  &:before {
+    display: block;
+    top: 0;
+    left: 0;
+    position: absolute;
+    content: "?";
+    z-index: 2;
+    width: 14px;
+    height: 14px;
+    border-radius: 50%;
+    line-height: 15px;
+    text-align: center;
+    vertical-align: middle;
+    font-family: sans-serif;
+    font-size: 12px;
+    font-weight: bold;
+    color: white;
+    background-color: ${colors.grayDark};
+  }
+  &:hover {
+    &:before {
+      z-index: 4;
+    }
+    &:after {
+      opacity: 1;
+      z-index: 3;
+      transition: opacity 0.125s;
+      transition-delay: 0.0625s;
+      visibility: visible;
+    }
+  }
+  &:after {
+    display: block;
+    top: 0;
+    left: 0;
+    position: absolute;
+    visibility: hidden;
+    text-align: left;
+    z-index: 1;
+    margin: -10px 0 0 -10px;
+    width: 300px;
+    color: ${colors.grayMedium};
+    background-color: white;
+    padding: 30px;
+    opacity: 0;
+    line-height: 1.3;
+    transform: translateZ(0);
+    border: 1px solid ${colors.grayDark};
+    content: ${(props: Props) => `"` + props.message + `"`};
```

Comment:
This took a little while to figure out, have to type it so the additional `message` prop is available to TS, and also add the quotes.

Also most likely width, color, and some positioning should be props that can be passed down, but since this is the first tooltip, I just went with the specific size needed for the market insights tooltip. If/when we need a tweaked tooltip we can make that refactor.

Reply:
It’s possible to properly type the props of the resulting React component, as shown here https://github.com/styled-components/styled-components/issues/630#issuecomment-315475109. This technique is already used in Reaction’s codebase so there should be plenty examples.

Reply:
Thanks, works great!

Reply:
🎩 👌